packages/editor/__tests__/model/appModel.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { AppModel } from '../../src/AppModel/AppModel';
+import { ComponentId, ComponentType } from '../../src/AppModel/IAppModel';
+import { AppSchema, DuplicatedIdSchema } from './mock';
+
+describe('AppModel test', () => {
+  const appModel = new AppModel(AppSchema.spec.components);
+  describe('resolve tree', () => {
+    it('resolve Tree corectlly', () => {
+      expect(appModel.allComponents.length).toBe(10);
+      expect(appModel.topComponents.length).toBe(2);
+    });
+
+    it('components order', () => {
+      expect(appModel.allComponents[0].id).toBe('hstack1');
```

Comment:
maybe `expect(appModel.allComponents.map(c => c.id)).toMatchInlineSnapshot()`

Reply:
I use `toEqual` instead. I think it is clearer.